pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -3040,7 +3040,8 @@ protected void handleCommandWatchTopicList(CommandWatchTopicList commandWatchTop
         final NamespaceName namespaceName = NamespaceName.get(commandWatchTopicList.getNamespace());
 
         Pattern topicsPattern = Pattern.compile(commandWatchTopicList.hasTopicsPattern()
-                ? commandWatchTopicList.getTopicsPattern() : TopicList.ALL_TOPICS_PATTERN);
+                ? Pattern.quote(commandWatchTopicList.getTopicsPattern())
```

Comment:
Isn't `Pattern.quote` going to disable any wildcard matches and regex behavior?

Reply:
Yep, It should be a breaking changes here. :) 

```java
class Scratch {
    public static void main(String[] args) {
        final String regex = "a*b";
        final String quoteRegex = Pattern.quote(regex);
        final Pattern pattern1 = Pattern.compile(regex);
        final Pattern pattern2 = Pattern.compile(quoteRegex);
        final Matcher a = pattern1.matcher("aaab");
        System.out.println(a.find());
        final Matcher b = pattern2.matcher("aaab");
        System.out.println(b.find());
    }
}

```
true
false
```

Reply:
Ouch. Let's close this then and focus on using re2 library